### DISCUSSION TITLE
ssh -T git@github.com with -oHostKeyAlgorithms=ssh-ed25519 doesn't reply welcome message

### DISCUSSION THREAD

**Discussion by imba-tjd** [community/community#50925] [Projects and Issues] [answered]
### Select Topic Area

Bug

### Body

```bash
> ssh -T git@github.com
Hi imba-tjd! You've successfully authenticated, but GitHub does not provide shell access.

> ssh -T git@github.com -oHostKeyAlgorithms=ssh-ed25519
(empty, but I can normally use git)
```

Tested on Win11 OpenSSH_for_Windows_8.6p1, LibreSSL 3.4.3

**Comment by airtower-luna** [comment, marked as answer]
Try adding `-v` to the command line to get debug output and hopefully see what's wrong. I can say from experience that using `ssh-ed25519` for the host key works.

**Reply by imba-tjd** [reply]
As I understand, HostKeyAlgorithms is used to ask for server fingerprint.

**Reply by airtower-luna** [reply]
Highly simplified, yes. It says which kind of key you're willing to accept from the server, and the server has a different key (which also means a different fingerprint) for each key type it supports.

But you'll need the debug output to see why the command line option to change the type isn't working for you.

**Reply by imba-tjd** [reply]
Thanks. You are right.
I used -v and found the Server host key is ssh-ed25519 even if I didn't set HostKeyAlgorithms.
I then used -vvv and found the log didn't helpful. Most of them are the same as not set. The last line makes no sense, the failure reason isn't logged.
So I downloaded the new version from https://github.com/PowerShell/Win32-OpenSSH/releases and it works......

The tricky thing is that I can reproduce the failure when testing GitHub. And it works on GitLab.
